### PR TITLE
Add support for pagination, sorting and filtering

### LINKED ISSUE #12: Enable Pagination
Enable Pagination for the list of users in the UI.

### LINKED ISSUE #14: Add ability to sort, filter by Project ID
This is requested by the clinical sites project admins for easily checking the authorised participants.

### PR DESCRIPTION
**Pull request by yatharthranjan**
Adds support for Client side pagination, sorting and filtering.
- 

NOTE: For better performance this should be implemented in the future in backend. #17 

- Closes #14 
- Closes #12 
- Adds pagination, sorting and filtering using mat-table, mat-paginator and mat-form-field
- One form for filtering for any column in the table
- Adds material theme

Here is a gif of how it looks -

radar-auth-example

### CODE REVIEW

File: authorizer-app/src/app/components/rest-source-authorization/rest-source-user-list.component.html [hunk 1]
```diff
@@ -1,40 +1,74 @@
...
+            <button mat-raised-button	color="warn" (click)="removeDevice(row)">Delete</button>
+        </mat-cell>
+      </ng-container>
+
+      <mat-header-row *matHeaderRowDef="displayedColumns; sticky: true"></mat-header-row>
+      <mat-row *matRowDef="let row; columns: displayedColumns;">
+      </mat-row>
+    </mat-table>
+
+  <mat-paginator [pageSize]="5" [pageSizeOptions]="[5, 10, 20, 50]" showFirstLastButtons></mat-paginator>
```
Comment (blootsvoets) [inline authorizer-app/src/app/components/rest-source-authorization/rest-source-user-list.component.html]: Perhaps set the default to 20 per page?

**Comment by nivemaham** [comment]
Nice work. Although I prefer to have the pagination supported from back-end as well. Pagination with Spring is fairly straight forward. Could you take that up @yatharthranjan

**Comment by yatharthranjan** [comment]
@blootsvoets No updated needed for `package.json`. 

@nivemaham I looked into it when I started implementing this and I think it will require some work and probably for another PR. As it will involve adding all the 3 (pagination, sorting and filtering) from the backend and then also creating custom datasources in the frontend as `MatTableDataSource` only supports client side pagination, sorting and filtering. For the front end part, see this post for an example -- https://blog.angular-university.io/angular-material-data-table/ 
I will create an issue for this and maybe try to do it another PR but this should be sufficient for now with the number of rows we have

**Comment by yatharthranjan** [comment]
Hi Guys can we proceed with this please ? I have created issue for server side support #17

**Comment by yatharthranjan** [comment]
Yes that is because I ran `npm update` and caused updated versions in `package-lock.json` for dependencies with Caret `^` and Tilde `~` in `package.json` ?

**Pull request merged**